torch/_inductor/codegen/cpp.py

```diff
@@ -3653,30 +3653,36 @@ def _can_fuse_nodes_with_compatible_ranges(self, node1, node2):
         if isinstance(node_to_recomp, FusedSchedulerNode):
             return False
 
-        def get_buffer(node):
-            if isinstance(node, FusedSchedulerNode):
-                assert len(node.snodes) > 0
-                # use the last scheduler node from the list as it has the most
-                # relevant indexing expressions
-                return get_buffer(node.snodes[-1])
-            else:
-                assert isinstance(node, SchedulerNode)
-                return node.node
-
-        ref_node_buffer = get_buffer(ref_node)
-        if isinstance(ref_node_buffer, ir.TemplateBuffer):
-            return False
-
-        assert isinstance(ref_node_buffer, ir.ComputedBuffer)
-
         # It may happen that node1 and node2 compatible number of elements
         # but different original ranges, for example:
         # {d0: s0, d1: s1, d2: s2} vs {d0: s0*s1*s2}
         # See https://github.com/pytorch/pytorch/pull/120077/files#r1500427848 for more details
         # TODO: we can fix if it allows us to CSE at least one of the variables
-        var_ranges1 = ref_node_buffer.get_read_writes().var_ranges
-        var_ranges2 = node_to_recomp.node.get_read_writes().var_ranges
-        if var_ranges1 != var_ranges2:
+
+        assert isinstance(node_to_recomp, SchedulerNode)
+        if isinstance(node_to_recomp.node, ir.TemplateBuffer):
+            return False
+        assert isinstance(node_to_recomp.node, ir.ComputedBuffer)
+        # node.data.get_size() is a cheaper version of node.get_read_writes().var_ranges
+        # but without variable name
+        ranges2 = node_to_recomp.node.data.get_size()
+        ranges1 = None
+        if isinstance(ref_node, FusedSchedulerNode):
+            ranges_list = []
+            for snode in ref_node.snodes:
+                if isinstance(snode.node, ir.TemplateBuffer):
+                    break
+                assert isinstance(snode.node, ir.ComputedBuffer)
+                ranges_list.append(tuple(snode.node.data.get_size()))
+
+            if len(set(ranges_list)) == 1:
+                ranges1 = list(ranges_list[0])
```

Comment:
more like: If it's not length 1 -> return false, right? But also, what's an example where this would not have length 1?

Also2, list(ranges_list[0])??

Reply:
if length is not 1, than `ranges1` is None and below it wont match ranges2 and thus return false.
I do not know right now any cases when FusedSchedulerNode could contain nodes with different ranges. But this is related to the above assumption https://github.com/pytorch/pytorch/pull/122420#discussion_r1535628050 and also the fact that FusedSchedulerNode does not define its read_writes with non-None var_ranges vs how it is done for SchedulerNode (see dependencies.extract_read_writes).

Reply:
I mean, sure, but It's cleaner to write
```suggestion
            if len(ranges_set) != 1:
                return False
            ranges1 = next(iter(ranges_set))
```